### PR TITLE
Edits to use Dask rather than multiprocessing for internal parallelism

### PR DESCRIPTION
**Pull request by lfrank**
These edits allow parallel execution in SpikeInterface to work

### CODE REVIEW

**Comment by magland** [comment]
This is great @lfrank! Could you also add "dask" to the install_requires in setup.py?

**Comment by lfrank** [comment]
I think that should now be added.
--
Loren M. Frank

Professor  
Howard Hughes Medical Institute
Kavli Institute for Fundamental Neuroscience
Department of Physiology 
University of California, San Francisco
Office Phone: 415-502-7357

**Comment by magland** [comment]
We need a , after 'dask'
@lfrank

**Comment by lfrank** [comment]
Let me know if there is anything else...
--
Loren M. Frank

Professor  
Howard Hughes Medical Institute
Kavli Institute for Fundamental Neuroscience
Department of Physiology 
University of California, San Francisco
Office Phone: 415-502-7357

File: ml_ms4alg/ms4alg.py [hunk 1]
```diff
@@ -685,27 +693,27 @@ def sort(self):
...
-        pool.close()
-        pool.join()
         #for m in range(M):
         #    print ('Running phase1 neighborhood sort for channel {} of {}...'.format(m+1,M)); sys.stdout.flush()
         #    neighborhood_sorters[m].runPhase1Sort()
-
+        dask_list = []
         for m in range(M):
             times_m=neighborhood_sorters[m].getPhase1Times()
             channel_assignments_m=neighborhood_sorters[m].getPhase1ChannelAssignments()
```
Comment (wysota) [inline ml_ms4alg/ms4alg.py:707]:
```suggestion
        for sorter in neighborhood_sorters:
            times_m=sorter.getPhase1Times()
            channel_assignments_m=sorter.getPhase1ChannelAssignments()
```

File: ml_ms4alg/ms4alg.py [hunk 2]
```diff
@@ -685,27 +693,27 @@ def sort(self):
...
             for m2 in range(M):
                 inds_m_m2=np.where(channel_assignments_m==m2)[0]
                 if len(inds_m_m2)>0:
                     neighborhood_sorters[m2].addAssignedEventTimes(times_m[inds_m_m2])
+        for m in range(M):
+            tmp_dask = dask.delayed(run_phase2_sort(neighborhood_sorters[m]))
```
Comment (wysota) [inline ml_ms4alg/ms4alg.py:713]:
```suggestion
        for m in neighborhood_sorters[:M]:
            tmp_dask = dask.delayed(run_phase2_sort(sorter))
```

Comment (wysota) [reply ml_ms4alg/ms4alg.py:713]: Or replace `neightborhood_sorters[:M]` with `itertools.islice(neighborhood, M)` if you expect M to be large and care about performance.

**Pull request merged**